proto/api.proto

```diff
@@ -0,0 +1,315 @@
+};
+
+// CN--->DN, DDL
+// create database test: 
+//      Entry {
+//          entry_type = Insert
+//          table_id = 0
+//          database_id = 0
+//          table_name = mo_database
+//          database_name = mo_catalog
+//          file_name = "" 
+//          block_id = 0
+//          bat = "test, 0, ..." 
+//      }
+// drop database test:
+//      Entry {
+//          entry_type = Delete
+//          table_id = 0
+//          database_id = 0
+//          table_name = mo_database
+//          database_name = mo_catalog
+//          file_name = "" 
+//          block_id = 0
+//          bat= "test, 0, ..." 
+//      }
+
+// create a user table: 
+//  insert entry into mo_tables and ...;
+//      Entry {
+//          entry_type = Insert
+//          table_id = 1
+//          database_id =  0
+//          table_name =  mo_tables
+//          database_name = mo_catalog
+//          file_name = "" 
+//          block_id = 0
+//          bat.attrs = {"relid", "relname", "reldatabase", ...}
+//      }
+
+// drop table: 
+//  delete entry from mo_tables and ...;
+//      Entry {
+//          entry_type = delete
+//          table_id = 1
+//          database_id = 0
+//          table_name =  mo_tables
+//          database_name = mo_catalog
+//          file_name = "" 
+//          block_id = 0
+//          bat.attrs = {"relid", "relname",...}
+//      }
+
+//  bulk load:
+// bulk loads a block into S3.
+//      Entry {
+//          entry_type = Insert
+//          table_id = tid
+//          database_id = dbId
+//          table_name =  xxx
+//          database_name = xxx
+//          file_name = "s3 file name" 
+//          //transient block id;
+//          block_id = "xxx" 
+
+//          //block meta info and colunm meta info
+//          bat.attrs = {"clounm index", "min", "max", "bf location", "data location",...}
+//      }
+
+//  DML:
+// append a batch of data into table;
+//            Entry {
+    //          entry_type = Insert
+    //          table_id =  tid
+    //          database_id = dbId
+    //          table_name =  xxx
+    //          database_name = xxx
+    //          file_name = "" 
+    //          block_id = 0
+    //          bat.attrs = {"primary column value",  "column1 value", ...}
+    //      }
+
+//  delete batch of data from table;
+//            Entry {
+    //          entry_type = Insert
+    //          table_id =  tid
+    //          database_id = dbId
+    //          table_name =  xxx
+    //          database_name = xxx
+    //          file_name = "" 
+    //          block_id = 0
+    //          bat.attrs = {"rowid", "primary column value",  "column1 value", ...}
+    //      }
+
+
+message Entry {
+    enum EntryType {
+        Insert = 0;
+        Delete = 1;
+    }
+    EntryType entry_type = 1;
+    uint64    table_id = 2;
+    uint64    database_id = 3;
+    string    table_name = 4;
+    string    database_name = 5;
+    string    file_name = 6;
+    uint64    block_id = 7;
```

Comment:
Entry maybe Batch of tuples, or, a block.   The block should contain a field  that has the same format as a BlockMeta used below.